bzzeth/bzzeth_test.go

```diff
@@ -223,3 +260,170 @@ func isPeerDisconnected(id enode.ID, b *BzzEth) (p *Peer) {
 	}
 	return
 }
+
+func newBlockHeaderExchange(tester *p2ptest.ProtocolTester, peerID enode.ID, requestID uint32, offered *NewBlockHeaders, wanted [][]byte) error {
+	return tester.TestExchanges(
+		p2ptest.Exchange{
+			Label: "NewBlockHeaders",
+			Triggers: []p2ptest.Trigger{
+				{
+					Code: 1,
+					Msg:  offered,
+					Peer: peerID,
+				},
+			},
+			Expects: []p2ptest.Expect{
+				{
+					Code: 2,
+					Msg: GetBlockHeaders{
+						ID:     requestID,
+						Hashes: wanted,
+					},
+					Peer: peerID,
+				},
+			},
+		})
+}
+
+func blockHeaderExchange(tester *p2ptest.ProtocolTester, peerID enode.ID, requestID uint32, wantedData []rlp.RawValue) error {
+	return tester.TestExchanges(
+		p2ptest.Exchange{
+			Label: "BlockHeaders",
+			Triggers: []p2ptest.Trigger{
+				{
+					Code: 3,
+					Msg: BlockHeaders{
+						ID:      requestID,
+						Headers: wantedData,
+					},
+					Peer: peerID,
+				},
+			},
+		})
+}
+
+// Test bzzeth full eth node sends new block header hashes
+// respond with a GetBlockHeaders requesting headers falling into the proximity of this node
+// Also test two other conditions
+// - If a header is already present in localstore, dont request it in GetBlockHeaders
+// - If a unsolicited header is received, dont store it on localstore
+// Apart from that it also tests if all the headers are delivered and stored in localstore
+func TestNewBlockHeaders(t *testing.T) {
+	prvKey, netstore, cleanup := newTestNetworkStore(t)
+	defer cleanup()
+
+	// bzz pivot - full eth node peer
+	// NewBlockHeaders trigger, expect
+	tester, _, teardown, err := newBzzEthTester(t, prvKey, netstore)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer teardown()
+
+	offered := make(NewBlockHeaders, 256)
```

Comment:
candidate for table driven test with different numbers of block headers

Reply:
Table is not possible here as we use bulk arguments in the messages...